### PR TITLE
HH-300 Add dependency check

### PR DESCRIPTION
**Pull request by kristianjokela**
## Description
Add dependency check (https://github.com/bmish/check-dependency-version-consistency) to project and docs how to use. The deps are checked and updated with the tool in this PR too.

For now you should run this from time to time, but this could/should also added to pre-commit hook or somewhere in CI pipeline.

### Closes

**[HH-300](https://helsinkisolutionoffice.atlassian.net/browse/HH-300):**

## Additional notes

[HH-300]: https://helsinkisolutionoffice.atlassian.net/browse/HH-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr268.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr268.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr268.dev.hel.ninja :laughing::tada::tada::tada:

**Review by nenonja — COMMENTED** [review]
Looks good and nice to have also all the versions bumped as well! I tried it and seemed to work when I downgraded some package. It was also very fast check.
The only comment is that I would probably add this check to the husky lint-staged config so that the feedback would be always instant, and not forgotten.
After merge perhaps @terovirtanen can add this to be a part of the Ci checks?

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr268.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr268.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr268.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr268.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by kristianjokela** [comment]
> Looks good and nice to have also all the versions bumped as well! I tried it and seemed to work when I downgraded some package. It was also very fast check. The only comment is that I would probably add this check to the husky lint-staged config so that the feedback would be always instant, and not forgotten. After merge perhaps @terovirtanen can add this to be a part of the Ci checks?

👍 I added the check: https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/268/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R27

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "g:lint": "yarn workspaces foreach -ptv run lint --color",
     "g:lint-styles": "yarn workspaces foreach -ptv run lint-styles --color",
     "g:typecheck": "yarn workspaces foreach -ptv run typecheck",
-    "g:lint-staged-files": "lint-staged --allow-empty",
+    "g:lint-staged-files": "lint-staged --allow-empty --debug && yarn lint:dependency-versions",
```
Comment (nenonja) [inline package.json:27]: For some reason I couldnt make it to work? I changed the root package with `"next": "12.3.4"` then made a commit about it and expected it to fail but it didnt. 
Perhaps scripts should be left as they are but instead add a new rule into 
https://github.com/City-of-Helsinki/events-helsinki-monorepo/blob/main/packages/eslint-config-bases/lint-staged.config.js#L18
For example when any of the package.json or yarn.lock is changed then run the check

**Review by nenonja — COMMENTED** [review]
Still one more suggestion :)

File: package.json (hunk 1 above)
Comment (nenonja) [reply package.json:27]: Perhaps this isnt so easy I first though :) If it's too hard to get work then perhaps in another pr.

Comment (kristianjokela) [reply package.json:27]: Yeah it doesn't actually fail; it just shows the boxes with dep diff; so as last like after all the other output from lint-staged, there should be the "boxes" from dep check that just show that there's a difference.

Comment (nenonja) [reply package.json:27]: Actually there was a proablem on my machine and it actually works! So I can approve this now

**Review by nenonja — APPROVED** [review]
Looks good to me!

**Pull request merged**